### ISSUE TITLE
[Feature Request] Option for (semi-)automatic updates

### ISSUE THREAD

**Issue by seafox** [manjaro/pamac#216]
Now when pamac got rid of pacman and its AUR helpers dependencies, it would be a logical usability step to allow an option for automatic updates. Or semi-automatic, if the current user has to be asked for sudo password OR just wish to manually control when it should be performed. An extra context menu 'Update Now' in for the app tray icon is desirable. Preferably with the default left click action as:
a) Open Package manager (pamac main UI) when system is up-to-date
b) Performing an Update Now action when there are pending updates.

**Comment by guinux** [comment]
You can already update your system with two clicks: one on the tray-icon to launch pamac-updater and one on "Apply". The benefits of what you describe would be limited.

**Comment by seafox** [comment]
Sure, they would be limited to those lazy folks who prefer (normally) service-free Windows style updates. Guess what percentage of regular Manjaro users that would be?

**Comment by polarathene** [comment]
While I can see it being a nice feature, I've had multiple updates in the past year on a few systems which required user intervention else the update would fail or your system/software breaks :\

**Comment by seafox** [comment]
Indeed, there will be times when autoupdate will fail or even break the system. But, in essence, exactly the same thing will happen when user just press Apply button when pamac indicator turns red.
That will happen in some % of cases (1-2 %?). To effectively prevent such situations in advance, Arch/Manjaro maintainers should issue a auto-update BLOCKING warning propagated directly to software (pamac). Something like notification message in kalu, but automated in a way that (semi)automatic update or even just regular manual pressing Apply button will be blocked in case there are breaking updates requiring user intervention. So only after the message with instructions to proceed is displayed and confirmed, user would be able to either try to perform the rourine system update or cancel and proceed in alternative way.

**Comment by seafox** [comment]
When thinking further about a feasible implementation scenario, there the need to provide clearly documented message exchange protocol between (decentralized like DHT approch or just software installed on repository servers) authorized devs that can issue the system update BLOCKing messages and users' pamac clients. The user would have an opt-out option, but by default will submit a message as a feedback of Apply button (System Update) operation outcome. In simplest form, it may be only 1 bit (SUCCESS/FAILURE) and applied only to official repos (AUR excluded). Even that simplest feedback channel would provide Alert data for devs to monitor the system updatability status. That info should be propagated directly to the distribution home page as real-time stats. Thus users will notice if something goes unusually as well. More than that, the collected total count for predefined time interval may be sent back to pamac so users will be alerted right before taking decision. That way the proposed here warning system may be completely automated. After reaching preset threshold (amount failures per last X hours), pamac may display the warning dialog with latest message feed. Further improvement would be when packages involved in problematic update transactions would be automatically submitted by pamac, then (auto)analyzing the stats to pinpoint problematic package chain(s) for (auto-generated or approved) alerts. Folks?

**Comment by avrelaun** [comment]
There's definitely a need for (optional) automatic updates out there. Most non-technical people already either blindy install updates or worse, never install updates. For sure some knowledge should be spread about updating and maintaining an OS, but there must be a better way than requesting user input regardless of the importance / risk of the update. Here's a proposition of automation levels:

- No automation at all
- Check for updates and notify new updates available
- Check for updates, install those who do not require user input automatically, notify user of the others
- Check for updates, install all automatically, notify of important updates done with the default action executed

**Comment by MaycneSonahoz** [comment]
The request of full automatic updates for the sake of user-friendliness is a bad idea to me. Just considering that such feature is not "easily" enabled on any distro should give a hint. As far as i know, only OpenSUSE permit it; even Ubuntu restrict it to security updates.
I understand the temptation to have a system that goes entirely out of the way of the user. But i don't think a system that breaks because of an update, when it happens, without the user understanding why, since it's automated, is user-friendly. And i think we all agree that Arch derivatives are not known for their perfect stability...

As for signaling and blocking the update within Pamac, i don't see those being any useful, since the main package manager of Arch derivatives in not Pamac, but Pacman. Even regarding Manjaro, let's not forget the KDE variant comes with Octopi instead of Pamac by default.

**Comment by polarathene** [comment]
> Even regarding Manjaro, let's not forget the KDE variant comes with Octopi instead of Pamac by default.

So ship Pamac instead? I'm on KDE and switched to Pamac, I've got octopi which works around a few bugs Pamac has that devs are yet to fix(not being able to update AUR packages if I have a main repo package I choose not to update being a main pain).

If Calamares gets BTRFS support, the system can use that FS with XFS on home like like openSUSE has with their excellent BTRFS layout and Snapper integration. Add that support to Pamac/Pacman for pre and post updates and if any update goes bad the user rollsback(via boot selection or Pamac UI).

At the moment I have quite a few packages that will refuse to update due to expecting an older version of openSSL as the one Manjaro updated to a few weeks ago is still in Arch unstable. Not sure how auto updates would handle things like that.

**Comment by MaycneSonahoz** [comment]
If you want a GUI for BTRFS management, i think there are tools for that -- i have no name in mind. I don't think it is wise to integrate that to a package manager.
Since you're comparing with OpenSUSE, i suppose you're thinking about how Yast might behave (i don't know it much). But Yast is a system configuration tool, which includes a GUI for the Zypp package manager. Pamac is only a GUI for the Pacman package manager, and includes package management for AUR.

Regarding packages not updated due to a missing dependency version, i don't see why automatic updates would differ from regular ones. Dependency resolution failed, so the package manager denies the update. A package manager only updates to newer versions, so if you want to [downgrade](https://wiki.archlinux.org/index.php/downgrading_packages), it is up to you to do it, manually.

**Comment by polarathene** [comment]
> If you want a GUI for BTRFS management, i think there are tools for that -- i have no name in mind. I don't think it is wise to integrate that to a package manager.

I don't know if it needs to be integrated into the package manager, might be a thing Manjaro could do if they supported BTRFS with the installer. Snapper integration is through pre/post hooks on Pacman iirc. Not sure what openSUSE is doing to have the boot support but handy if an update goes bad preventing a working boot.

I bring this setup from openSUSE up because 1, you mention it as being the only one you know supporting it by default and 2, I've tried to explain why it works and how Manjaro could benefit from it to avoid the concern you stated:

> I understand the temptation to have a system that goes entirely out of the way of the user. But i don't think a system that breaks because of an update, when it happens, without the user understanding why, since it's automated, is user-friendly. And i think we all agree that Arch derivatives are not known for their perfect stability...

**Comment by seafox** [comment]
> 
>     No automation at all
>     Check for updates and notify new updates available
>     Check for updates, install those who do not require user input automatically, notify user of the others
>     Check for updates, install all automatically, notify of important updates done with the default action executed
> 
@avrelaun Agreed, I'd add the last one:
* Check for updates, install all automatically, skip any notifications (i.e. "silent & gratified" mode) if no error(s) occured.

**Comment by seafox** [comment]
@MaycneSonahoz 

> As for signaling and blocking the update within Pamac, i don't see those being any useful, since the main package manager of Arch derivatives in not Pamac, but Pacman. Even regarding Manjaro, let's not forget the KDE variant comes with Octopi instead of Pamac by default.
Linux has modular structure that is flexibly evolving according user choices and needs. One can choose alpm, pacman or wrapper like pacaur to perform day-to-day maintainance. Choices are users'. What I'm suggesting here is a _default_ automation for Arch package system based distro like Manjaro. None of managers you mentioned have any kind of automation, so I see no further discussion point here.

**Comment by MaycneSonahoz** [comment]
> you mention it as being the only one you know supporting it by default

I did say OpenSUSE permit it, but not by default. The user still needs to enable it.
But my point was about most distros _not_ permitting it. Even "user-friendly" distros like Elementary or Mint do not permit it.

> What I'm suggesting here is a default automation for Arch package system based distro like Manjaro. None of managers you mentioned have any kind of automation, so I see no further discussion point here.

I suppose the Manjaro team wants to bring the same user-friendliness to (at least) all their official variants (KDE, Xfce, GNOME). They made the choice of Octopi for KDE, so bringing automation (and update feedback) to only Pamac (because that's where we are) might leave 1/3 of their users out of the loop; unless they do decide to ship Pamac everywhere by default. This might be a broader change than you expect.

**Comment by seafox** [comment]
We have to distinguish two different approaches to automatic OS maintainance discussed here. That is regarding the (possible) conflicts resolution mechanisms involved.

The first one, as suggested by OP, based on pro-active method of resolving based on first testers experienced results that are the outcome of update operations performed on real systems. For them it's post factum (aposteriory) failure or success. For vast majority of their followers that would be apriory flags and warnings ready to consume/take into account. That system is indeed more complex and heavily based on Arch packaging system's and/or pamac/pacaur/etc. conflict resolution stuff. Also important to note, that rollbacks, including downgrading packages are the integral part of the arsenal to bring the system back on track. In that regard the current Manjaro tri-stage (unstable/testing/stable) separate repositories can be seen as an earlier attempt to resolve intrinsic Arch based system's challenges. So at some point Manjaro can have flexible "cutting edge" like ArchLinux package set that uses (appropriate to maintain up-to-date system) mix of packages from all 3 current repository sets. Say KDE parts may be from latest unstable branch, while some specific security related tools/libraries from stable branch.

The second mechanism suggested here is not pro-active but rather reactional. Indeed, if FS like BTRFS with the ability to keep snapshots may have deeper integration into system (exposing updates related snapshots as GRUB items to boot from) like SUSE. I'd argue such "always success to boot into _operational_ system" update strategy has two intrinsic flaws that can't be ignored:
1. It has logical flaw that it doesn't resolve actual problem of keeping the system up-to-date. Essentially it's just a rollback mechanism upon encontered failure. Instead of providing the actual resolution of best possible combination of packages for any given time point with all the packages versions to choose from (as is in case of pro-active system described above), it merely postpones the actual resolution to later point in time, when more changes in packages hopefully happen. It also usually leaves a mess for automation with regard to what and how many previous snapshots to keep in everything in order for potentially deeper rollback needs and to have minimal space waste.
2. Not all file systems have snapshots, there are still SW specific issues with BTRFS. Many HW devices limited in resources making usage of snapshot FS not feasible.

**Comment by seafox** [comment]
@MaycneSonahoz 

> I suppose the Manjaro team wants to bring the same user-friendliness to (at least) all their official variants (KDE, Xfce, GNOME). They made the choice of Octopi for KDE, so bringing automation (and update feedback) to only Pamac (because that's where we are) might leave 1/3 of their users out of the loop; unless they do decide to ship Pamac everywhere by default. This might be a broader change than you expect.

I'm using KDE with pamac and pacaur, also kalu and octopi are installed. I do all updates through either pacaur or pamac. Since starting using pamac regularily so far I didn't encountered the need for octopi. _For me_ It's just more clunky. I know it's better suited for KDE due to its underlying UI base. Anyway I see no reason Manjaro team would prefer to have different system (package) managers depending on currently active desktop UI in the long run.

**Comment by polarathene** [comment]
@seafox I'm not sure what your points on BTRFS snapshots is, do you understand it's functionality properly? Unlike other filesystems the snapshots take no additional space(only at a later date if you were to delete files that belong to the snapshot will they still use disk space until the snapshot references are cleared and cleanup is done). 

If done like openSUSE you only do this on system data, not user data(/home is avoided). openSUSE has Snapper setup to only keep X snapshots in number or time iirc, so that's not an issue either. The fact that you could do an update and not be able to boot because of some package(possible with AUR that Manjaro team doesn't have responsibility over), is not a fun experience. Not sure how that isn't a useful and user friendly feature?

What SW issues are you referring to? RAID 5/6? It's not advised, but non parity modes are fine. It wouldn't be the default system FS for openSUSE for so long if it was in such a problematic state. Nor would you have NAS products with commercial products/software built with it(Synology, Rockstor).

**Comment by seafox** [comment]
@polarathene I see you are a proponent of BTRFS. There isn't any magic with snapshots, if you change them, they start to take space. We wouldn't discuss the _efficiency_ or _feasibility_ of different FSs here anyway. IMHO requirement that /home dir has different FS or container isn't a strength or advantage either. Neither I did say there is no use cases where snapshots would provide great advantages...
If we couldn't boot after AUR package changes, then well... I'd say the auxilary AUR and core Arch/Manjaro/... repos division became rather artificial. Technically AUR need builds vs pre-built official repos, but that shouldn't be the only difference. Maybe then folks should re-think and re-partition the repos?
BTRFS had problems right from the GRUB not able to store user's last used OS choice, had some performance problems, may still have problems with transparent disk encryption and other limitations that I can't recall right now. NAS products/software is the specific domain to discuss here. Back to the subject, IMHO the two negative points of snapshot FS mentioned are still valid.

**Comment by MaycneSonahoz** [comment]
I think we all agree that the risk of system breakage from an automated -- or more accurately, unattended -- update on an Arch derivative is too high.
The main discussion lately was about how to make it safe, from signaling back issues and blocking the current update, to reverting a snapshot of packages or a file system. Both cases require a significant change on update process and/or default system choices, with success expected only in limited cases.

I remind you the goal here is, in the end, so users don't have to click two buttons once every few days. This amount of change is not worth it.

**Comment by seafox** [comment]
@MaycneSonahoz 
I wouldn't call the procedures described here as "unattended". Signalling/blocking all or some packages from update involves either manual/(semi)automatic hint/flag/stats forwarding to clients or feedback/stats sending from clients. In case rollback required on snapshot the different user action is also required during boot.

I'd like to see your statistical evidence of "_with success expected only in limited cases_". I'd suspect your clause quoted above is rather dramatic and in reality the failure rate will be a few percents max. Varying on a use case, frequency of update checks, choosen repo branch etc. I'd say on Arch with update checks every 1 to 6 hours (as security patches still have NO PUSH mechanism!) expecting users to bring to front updater (one click), wait for fetching a few seconds (step 2), then blindly pressing Apply button (step 3), then waiting and praying for successful updates (step 4), just to make sure the system has the latest security updates and to be able to close updater window (step 5) isn't something providing great UX of modern OS.

IMHO the proposed updating mechanisms are possible to implement incrementally.
Say first adding DHT distributed store accumulating received feedback from pamac on every branch-package-version update cycle (say limited to one send per client per 10-60 minutes max to include possible retries), with silently presenting current forwarded stats related to the packages needed to update (in step 2 mentioned above) would help a lot in assessing the probability of successful update just before pressing Apply button.

**Comment by sudoquai** [comment]
Full Autoupdate should be available as an optional feature for the following reasons:

- If you click on update in pamac you don't have the possibility for user intervention. An autoupdate has the same behaviour.
- We can use snapshots on btrfs enabled systems to prevent system breaks
- Some linux systems are already running virtualized and snapshots can be made easily. If something goes wrong we can easily revert the changes.

If this optional feature is added we can warn the user easily about the risc of enabling it. AUR support is also available.

**Comment by seafox** [comment]
After about a year of using archlinux with autoupdate scheduled a few times a day task with `pacaur` that lately has been replaced by `trizen` like this
`/usr/bin/trizen -Syu --noconfirm [--ignore=...]` 

Here my experience:

- Updates mostly (in more than 90+ %) succeeded completely in background, user sees only pamac tray indicator color changes, that tells the system has the potential to greatly simplify users life and free them from ape-like routine maintanance
- When machine is rebooted or shutdown, there is a need to either delay it with notification on update in process, or present a dialog asking to either rollback/stop the current updates and proceed with shutdown, or let it do it after successful update (delayed)
- Two typical problem cases may be semi-automated: lookup for a package unknown pgp signature key, present its data/owner and let user an option to add it to the keyring and proceed or skip the package installaton; and, the second case, let user a chance to remove local (usually library) files that prevent successful package update or skip the update.

**Comment by polarathene** [comment]
@seafox Good to know. I thoought if you reboot/shutdown while an update is happening that it wouldn't apply? At least when doing updates if something fails it doesn't seem to be applied(might just be specific package with prior ones having been updated? which I guess can risk breakage when some dependencies are updated but others or packages that use the older dependencies until updated do not update). 

openSUSE has the BTRFS pre/post snapshot with snapper which is nice and could be used with Manjaro(Calamares doesn't support BTRFS yet I think?). They recently introduced "transactional updates" which treats the root file system as read-only and /etc with overlayfs. This new approach creates a single snapshot with all the new updates applied if the update process completes successfully. However any package install or update exists in that new snapshot, the running/active session doesn't have access to it like the more traditional snapper approach with a read/write root file system, so you need to reboot the system to boot into the new read-only snapshot. 

It solves the auto update problem(a systemd service called reboot manager schedules the updates and reboot in early hours) pretty well but I don't know how I feel about having to reboot just to use a package I installed/updated. It does have the benefit that even when updating kernels/drivers you shouldn't experience breakage like you can currently with Manjaro until a reboot. It's usually advised for system updates with user packages(services or apps) being installed with Flatpak/Snap/Docker which would not require reboots to use.

Your pgp/gpg key issue can be avoided, I'm not sure if trizen supports it but one of the aur helpers at least does. They can either prompt about the problem and let you allow adding the key which it will handle automatically, or you can have it auto add the keys(security risk).

---

An autoupdate option would be nice, supporting user notification or delaying reboot/shutdown should be doable during update process. You could also perhaps schedule a time where it's less likely for the user to shutdown or reboot. Or just auto download to a cache(perhaps one that could persist after a reboot so an update can occur after?

Being able to pause/stop safely is useful too, especially if compiling AUR packages and it puts load on the system and the user needs to put the process on hold for a bit to use CPU resources.

**Comment by seafox** [comment]
IMHO there are fundamental issues with snapshot or read-only state plus reboot updating systems:

1. This introduce an extra level of complexity. As a snapshot updating has its obstacles and majority of distros do not have it yet for various reasons, generalization doesn't help.
2. It goes against the principle and spirit of live rolling distros as it doesn't try to resolve the problems in interactive or semi-automatic ways for known failure patterns, but instead just reverts back and postpone a resolution.
3. I think unexpected/unanticipated reboot or shutdown should NEVER be an option, as it KILLS the usability and even may cause a significant damage (depending on user's workflow). So no, please, never Windows style updates with interrupted control and idiotic reboots again! Especially for desktop installations in laptops, such behaviour should be considered as ill designed and plain rude.
4. The risk of breakage on forced shutdown is unfortunately still there. File system recovery messages shown in console on boot after such shutdowns are begging for at least presenting the user a warning before performing such shutdown, or, better yet, to have a minimal like 3...30 seconds delay to cache pending updates and resume them after reboot. 

As to other mentioned things. Adding gpg key automatically is indeed a security risk, so I suggested quering from a trusted server and presenting all available information about the key to the user. At least it's better than blindly accepting everything.

As to continuing updating from the cache, I think it's already the case, only hash checks performed to skip re-downloading the stuff.

And yes, ideally, the design of updaters should include an atomic workflow for ability to stop/pause/recover/continue update process, including FS checks and complete cleanup of the unfinished/uncommitted step(s).

**Comment by seafox** [comment]
And of course, when user chooses to cancel a package update on interactive updater's prompt, the package should be automatically placed in excusion list to ignore on consequent retries. Better yet, that should be placed on a separate list of Recently Failed Updates, to simplify a retry in a batch later. The normal ignore list should be considered as a (more) permanent one.

**Comment by guinux** [comment]
I think full auto update in background is not a good idea.
It's also a point that should be discussed in Manjaro development and not here.
However I'm working on downloading updates in background.

**Comment by seafox** [comment]
@guinux 
1. Please provide the link in Manjaro dev for continuing the discussion.
2. @notramo

> One problem with updating, that a dependency is updated but a program that requires it isn't yet updated. For example, blender depends on boost-libs. It is compiled against one version of boost-libs, and doesn't work with different version. Pamac starts updating, it updates boost-libs, then other packages, then blender. What happens if the user launches blender in that status of the update (boost-libs is updated, but blender isn't yet)? It will crash, because it finds a different version of Boost libs on the system than the one it was compiled against.
> Also, the user can get similar error, if a large package is currently in progress (for example libreoffice-fresh), and only part of the new version is extracted to the filesystem. It can crash in that case also.
> 
> What about when the user shuts down the machine, because he doesn't know that a transaction is in progress? Also, it can be a problem if the user supends the machine while updating. It maybe can't wake up.
> 
> Why I think that downloading only would be a good solution:
> If the user shuts down the machine while downloading, the next update will download again the last file that isn't finished, then continues with other packages.
> When a laptop is on low battery level (or there is some reason for shutting down the machine quickly), the user can choose not to update.
> The user will always know the updated packages and not wonder if some programs doesn't start while updating.

Although these points are somewhat valid, I'd suggest you to understand the real situation whenever ANY updating process happes: Irrespectively was it initiated by hand or on schedule, any of the problems described above still could happen! Thus, the only solution to perform reliable updates is need for a transactional updating mechanism. Any update could be interrupted by power failure, forced reboot etc., so there should be built-in service that does check for pending unfinished updates before GUI is ready for user interaction, perform the last step recognision and possibly needed cleanup (including syncing versions like rollback library updates in case user choose to postpone the updating after boot, or automatically, if core packages involved and there is the risk GUI will fail to start), then ask user permission to retry with updates or postpone.

> 
I think automatically compiling AUR packages can be problematic on old machines, because it requires almost all the CPU, and sometimes a lot of RAM. If somebody has a running browser, and office, the compiling can freeze their system.

As users still do want keep AUR packages up to date, there are few options to deal with that combined:
A. Choose low user interaction period (like idle time or web reading) and CPU load period for scheduled updates;
B. Find a way to share CPU cycles and limit resource consumption by compiler in case of significant CPU load (the threshold value [%] can be configured in AUR Update options)
C. Using suggested transactional update mechanism dealy next atomic step until load is moderate or idle state is detected.

**Comment by philmmanjaro** [comment]
partly added to **v6.3.91-1**

**Comment by seafox** [comment]
@philmmanjaro 
> partly added to v6.3.91-1
I'm on Arch and see nothing, the current AUR version is pamac 6.3.2-1. I think the AUR version shouldn't lay behind.
What parts exactly were added, only interruptable update feed by downloading in cache in background (so we don't see any extra options in preferences yet)? In AUR v6.3.x I started to see (excessively?) frequent re-checks for updates like even after very minor and quick single package updates and also after every activation / restoring pamac main window...
What functionality do you plan to implement if any?

**Comment by guinux** [comment]
It's not officially release now, I asked to add it in Manjaro unstable branch in order to have feedback.
In archlinux you can test it using pamac-aur-git.
There is a new option in settings to enable auto download updates.

**Comment by seafox** [comment]
@guinux 
> In archlinux you can test it using pamac-aur-git.
Then please remove pacman < 5.1 dependency first )

**Comment by philmmanjaro** [comment]
@seafox: we don't maintain any AUR packages. Please ask the author. Our focus is on **Manjaro** and not on **Arch**.

**Comment by seafox** [comment]
Dependency is now fixed, so git version is also available...
So do you guys have plans to implement a system update daemon with transactional atomic updating mechanism, and get rid of pamac modal confirmation-only dialogs in favor of system notification messages aiming to achieve a seemless in-background updating process?
Thanks!

**Comment by seafox** [comment]
Dependency is now fixed, so git version is also available... but looks like it doesn't download in background, at least as the red tray icon shows updates, the downloading starts only after clicking Apply button. Automatically Download Updates option is selected. Or does the feature require reboot after installation?

Currently when Manjaro keeps the winner's place in distrowatch, do you guys have plans to implement a system update daemon with transactional atomic updating mechanism, and get rid of pamac modal confirmation-only dialogs in favor of system notification messages aiming to achieve a seemless in-background updating process?
Thanks!

**Comment by guinux** [comment]
Please use last commit with pacman 5.1 support and check the option in preferences.
Also know that downloading updates in background is done by pamac-tray.

**Comment by seafox** [comment]
sure I got them today
pacman 5.1 + pamac-aur-git v6.3.3.r3.g6204d96-1

![image](https://user-images.githubusercontent.com/9025690/40682389-1c4d83c2-637b-11e8-9915-cc2138312a7e.png)

**Comment by philmmanjaro** [comment]
@seafox: no you don't. It should be at least [v6.3.3.r5](https://github.com/manjaro/pamac/compare/v6.3.3...master) ...

**Comment by seafox** [comment]
> @seafox: no you don't. It should be at least v6.3.3.r5 ...

I see. But AUR isn't in sync with that yet. I guess I took @guinux worlds too seriously:

> In archlinux you can test it using pamac-aur-git.

Guess it's not possible to test in AUR yet. But my question remains unanswered:

> Currently when Manjaro keeps the winner's place in distrowatch, do you guys have plans to implement a system update daemon with transactional atomic updating mechanism, and get rid of pamac modal confirmation-only dialogs in favor of system notification messages aiming to achieve a seemless in-background updating process?
> Thanks!

![image ](https://user-images.githubusercontent.com/9025690/40700162-e1fd0754-63c7-11e8-8cca-61aa71f64644.png)

**Comment by philmmanjaro** [comment]
Simply do a re-install of **pamac-aur**, it will get the latest sources from git. With AUR packages you won't get always triggered when there is an update at upstream.
